Sources/BraintreeThreeDSecure/BTThreeDSecurePostBody.swift

```diff
@@ -0,0 +1,96 @@
+import Foundation
+
+/// The POST body for `v1/payment_methods/\(urlSafeNonce)/three_d_secure/lookup`
+struct BTThreeDSecurePostBody: Encodable {
+    
+    let accountType: String?
+    let additionalInfo: AdditionalInfo
+    let amount: String
+    let cardAdd: Bool?
+    let challengeRequested: Bool
+    let customFields: [String: String]?
+    let customer: Customer
+    let dataOnlyRequested: Bool
+    let dfReferenceID: String?
+    let exemptionRequested: Bool
+    let requestedExemptionType: String?
+    let requestedThreeDSecureVersion: String?
+
+    init(request: BTThreeDSecureRequest) {
+        self.requestedExemptionType = request.requestedExemptionType.stringValue
+        self.requestedThreeDSecureVersion = "2"
+        self.accountType = request.accountType.stringValue
+        self.dfReferenceID = request.dfReferenceID
+        self.dataOnlyRequested = request.dataOnlyRequested
+        self.challengeRequested = request.challengeRequested
+        self.amount = request.amount
+        self.exemptionRequested = request.exemptionRequested
+
+        if let customFields = request.customFields {
+            self.customFields = customFields
+        } else {
+            self.customFields = nil
+        }
+
+        if request.cardAddChallengeRequested {
+            self.cardAdd = true
+        } else {
+            self.cardAdd = nil
+        }
```

Comment:
Are we able to simplify this to something like `self.cardAdd = request.cardAddChallengeRequested ? true : nil`. We may be able to do the same for custom fields above.